vmware-base/src/main/java/com/cloud/hypervisor/vmware/mo/VirtualMachineMO.java

```diff
@@ -2412,6 +2419,23 @@ public void ensureScsiDeviceControllers(int count, int availableBusNum) throws E
         }
     }
 
+    private boolean isValidScsiDiskController(VirtualSCSIController scsiDiskController) {
+        if (scsiDiskController == null) {
+            return false;
+        }
+
+        List<Integer> scsiDiskDevicesOnController = scsiDiskController.getDevice();
+        if (scsiDiskDevicesOnController != null && scsiDiskDevicesOnController.size() >= (VmwareHelper.MAX_SUPPORTED_DEVICES_SCSI_CONTROLLER)) {
```

Comment:
Maybe also a null check for scsiDiskDevicesOnController?

Reply:
update

Reply:
> Maybe also a null check for scsiDiskDevicesOnController?

updated